Univa/rumcake
# Enter bootloader by double tapping reset button

## Applicable Changes



- [ ] Modification to behaviour of existing functionality (potential breaking change)
- [ ] Modification to existing public APIs (potential breaking change)
- [ ] User docs modification
- [ ] Rust toolchain updates
- [ ] Workflow change (GitHub Actions)
- [ ] Repository housekeeping (e.g. READMEs, licensing, templates, etc.)

## Related Issues



#5 

## Description of Changes

Adds the option to enter the bootloader with a double tap of the reset button. This is useful for entering the RP2040 UF2 bootloader, if the regular boot button is hard to reach.

In the current implementation, the field `bootloader_double_tap_reset` was added to the keyboard macro. If the `bootloader-double-tap-reset` feature is enabled, this field can be set and optionally change the default timeout value (200ms).

Like:
`#[keyboard(bootloader_double_tap_reset)]`
or
`#[keyboard(bootloader_double_tap_reset = <val>)]`

## Testing Instructions

1. Enable the feature in the Cargo.toml and add `bootloader_double_tap_reset` to the keyboard macro. 
2. Compile firmware and flash onto a RP2040.
3. Press the reset button two times in quick succession and check if the UF2 bootloader was entered.

## Check-list

- [ ] I have updated or added to the user docs where applicable
- [X] I have tested the changes on real hardware if applicable
- [X] The changes do not break compilations with other potentially related feature flags
- [X] Credit given where applicable
